be/src/storage/range.h

```diff
@@ -315,6 +325,36 @@ inline SparseRange<T> SparseRange<T>::intersection(const SparseRange<T>& rhs) co
     return result;
 }
 
+template <typename T>
+inline void SparseRange<T>::split(size_t expected_range_cnt, size_t chunk_size) {
+    if (size() < expected_range_cnt && span_size() > std::max(expected_range_cnt, chunk_size)) {
+        size_t expected_size_each_range = 0;
+        for (size_t i = 0; i < size(); ++i) {
+            expected_size_each_range += _ranges[i].span_size();
+        }
+
+        expected_size_each_range /= expected_range_cnt;
+        expected_size_each_range = std::max<size_t>(expected_size_each_range, 1);
+
+        std::vector<Range<T>> new_ranges;
+        for (auto range : _ranges) {
+            while (range.span_size() > expected_size_each_range) {
+                new_ranges.emplace_back(range.begin(), range.begin() + expected_size_each_range);
+                range = Range<T>(range.begin() + expected_size_each_range, range.end());
+            }
+            new_ranges.emplace_back(range);
+        }
+        std::swap(_ranges, new_ranges);
+        _is_normalized = false;
+    }
+}
+
+template <typename T>
+inline void SparseRange<T>::reverse() {
+    std::reverse(_ranges.begin(), _ranges.end());
+    _is_normalized = false;
```

Comment:
_is_normalized = true?

Reply:
normalize requires multiple ranges to be ordered.